Sources/Core/Client.swift

```diff
@@ -305,6 +242,394 @@ final class Client {
+
+            let pack = try Converter.fromChangePack(result.changePack)
+            try await doc.applyChangePack(pack: pack)
+
+            self.attachmentMap[doc.getKey()] = Attachment(doc: doc, isRealtimeSync: !isManualSync, peerPresenceMap: [String: PresenceInfo]())
+            self.runWatchLoop()
+
+            Logger.info("[AD] c:\"\(self.key))\" attaches d:\"\(doc.getKey())\"")
+
+            return doc
+        } catch {
+            Logger.error("Failed to request attach document(\(self.key)).", error: error)
+            throw error
+        }
+    }
+
+    /**
+     * `detach` detaches the given document from this client. It tells the
+     * server that this client will no longer synchronize the given document.
+     *
+     * To collect garbage things like CRDT tombstones left on the document, all
+     * the changes should be applied to other replicas before GC time. For this,
+     * if the document is no longer used by this client, it should be detached.
+     */
+    @discardableResult
+    public func detach(_ doc: Document) async throws -> Document {
+        guard self.isActive else {
+            throw YorkieError.clientNotActive(message: "\(self.key) is not active")
+        }
+
+        guard let clientID = self.id, let clientIDData = clientID.toData else {
+            throw YorkieError.unexpected(message: "Invalid client ID! [\(self.id ?? "nil")]")
+        }
+
+        var detachDocumentRequest = DetachDocumentRequest()
+        detachDocumentRequest.clientID = clientIDData
+        detachDocumentRequest.changePack = Converter.toChangePack(pack: await doc.createChangePack())
+
+        do {
+            let result = try await self.rpcClient.detachDocument(detachDocumentRequest)
+
+            let pack = try Converter.fromChangePack(result.changePack)
+            try await doc.applyChangePack(pack: pack)
+
+            self.attachmentMap.removeValue(forKey: doc.getKey())
+
+            self.runWatchLoop()
+
+            Logger.info("[DD] c:\"\(self.key)\" detaches d:\"\(doc.getKey())\"")
+
+            return doc
+        } catch {
+            Logger.error("Failed to request detach document(\(self.key)).", error: error)
+            throw error
+        }
+    }
+
+    /**
+     * `sync` pushes local changes of the attached documents to the server and
+     * receives changes of the remote replica from the server then apply them to
+     * local documents.
+     */
+    @discardableResult
+    public func sync() async throws -> [Document] {
+        let documents = self.attachmentMap.values.compactMap { $0.doc }
+
+        do {
+            try await withThrowingTaskGroup(of: Void.self) { group in
+                documents.forEach { document in
+                    group.addTask {
+                        try await self.syncInternal(document)
+                    }
+                }
+
+                for try await _ in group {}
```

Comment:
This code seems unnecessary for similar reasons as [this](https://github.com/yorkie-team/yorkie-ios-sdk/pull/31#discussion_r1016122446).

Reply:
That's for exception catch.

Reply:
I found a better function. waitForAll()